File: Data Structures & Algorithms/duplicate-integer/submission-3.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        for (int i = 0; i < nums.length;i++){
            for(int n = 0; n < nums.length -1;n++){
                if (n == i){
                    continue;
                }
                if(nums[i] == nums[n]){
                    return true;
                }
            }
        }


        return false;
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-3.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        comb = []
        nums.sort()
        def dfs(i,rem):
            if (rem == 0):
                res.append(comb[:])
                return
            for j in range(i,len(nums)):
                if rem - nums[j] < 0:
                    return
                comb.append(nums[j])
                dfs(j,rem - nums[j])
                comb.pop(len(comb) -1)
        dfs(0,target)
        return res